pkg/webhook/resources/storageclass/validator.go

```diff
@@ -38,13 +39,35 @@ func (v *storageClassValidator) Resource() types.Resource {
 }
 
 func (v *storageClassValidator) Create(request *types.Request, newObj runtime.Object) error {
-	return v.validateSetUniqueDefault(newObj)
+	err := v.validateSetUniqueDefault(newObj)
+	if err != nil {
+		return err
+	}
+
+	return v.validateDataLocality(newObj)
 }
 
 func (v *storageClassValidator) Update(request *types.Request, oldObj runtime.Object, newObj runtime.Object) error {
 	return v.validateSetUniqueDefault(newObj)
 }
 
+// Harvester rejects setting dataLocality as strict-local, because it makes volume non migrateable,
+// beside, strict-local volumes only could have one replica, it make Longhorn block node drain
+// https://longhorn.io/docs/1.5.3/references/settings/#node-drain-policy
+func (v *storageClassValidator) validateDataLocality(newObj runtime.Object) error {
+	sc := newObj.(*storagev1.StorageClass)
+	dataLocality, find := sc.Parameters[util.LonghornDataLocality]
+	if !find {
+		return nil
+	}
+
+	if longhorn.DataLocality(dataLocality) == longhorn.DataLocalityStrictLocal {
```

Comment:
nit: do we also want to check the value is allowed? I tried specify "test" "abc" and it also passes.

Reply:
Updated per suggestion, thanks.